### PR TITLE
Build using _GLIBCXX_DEBUG for 1 Travis job

### PR DESCRIPTION
**Pull request by bjodah**
### TODO
- [x] show that tests pass with _GLIBCXX_DEBUG, ensure success
- [x] then: temporarily reintroduce a bug with out-of-bounds access in std::vector, ensure failure
- [x] then: fix it again, ensure success

### CODE REVIEW

**Comment by bjodah** [comment]
Ok, so `_GLIBCXX_DEBUG` works with gcc 6 at least. What about this:

1. I merge this PR as one squashed commit, wait for travis build of master to finish.
2. I push a commit to master removing [this line](https://github.com/symengine/symengine/pull/1419/files#diff-97e2e3c68308a6044377595c72c26cedR99)

That way the cache for that particular job gets updated with object files compiled with `_GLIBCXX_DEBUG`. Sounds good?

**Comment by isuruf** [comment]
I've removed all the caches from travis-ci. Can you remove the cache deleting line and we can merge without any problem

File: bin/test_travis.sh [hunk 1]
```diff
@@ -95,7 +95,11 @@ if [[ "${CC}" == *"clang"* ]] && [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
         export CXXFLAGS=""
     fi
 else
-    export CXXFLAGS="-Werror"
+    if [[ "${CC}" == *"gcc"* ]] && [[ "${BUILD_TYPE}" == "Debug" ]] && [[ "${BUILD_SHARED_LIBS}" == "yes" ]] && [[ "${WITH_GCC_6}" == "yes" ]] && [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
```
Comment (isuruf) [inline bin/test_travis.sh]: Should we just use an env variable `USE_GLIBCXX_DEBUG` and set it in .travis.yml ?

**Comment by isuruf** [comment]
Thanks

**Pull request merged**